samples/openapi3/client/petstore/python-experimental/tests/test_http_signature.py

```diff
@@ -0,0 +1,164 @@
+        self.category.id = id_gen()
+        self.category.name = "dog"
+        self.tag = petstore_api.Tag()
+        self.tag.id = id_gen()
+        self.tag.name = "python-pet-tag"
+        self.pet = petstore_api.Pet(name="hello kity", photo_urls=["http://foo.bar.com/1", "http://foo.bar.com/2"])
+        self.pet.id = id_gen()
+        self.pet.status = "sold"
+        self.pet.category = self.category
+        self.pet.tags = [self.tag]
+
+    def setUpFiles(self):
+        self.test_file_dir = os.path.join(os.path.dirname(__file__), "..", "testfiles")
+        self.test_file_dir = os.path.realpath(self.test_file_dir)
+        if not os.path.exists(self.test_file_dir):
+            os.mkdir(self.test_file_dir)
+        self.rsa_key_path = os.path.join(self.test_file_dir, 'rsa.pem')
+
+    def test_http_signature(self):
+        with open(self.rsa_key_path, 'w') as f:
+          f.write(RSA_TEST_PRIVATE_KEY)
+
+        signing_cfg = signing.HttpSigningConfiguration(
+            key_id="my-key-id",
+            private_key_path=self.rsa_key_path,
+            signing_scheme=signing.SCHEME_HS2019,
+            signing_algorithm=signing.ALGORITHM_RSASSA_PKCS1v15,
+            signed_headers=[
+                signing.HEADER_REQUEST_TARGET,
+                signing.HEADER_CREATED,
+                signing.HEADER_HOST,
+                signing.HEADER_DATE,
+                signing.HEADER_DIGEST,
+                'Content-Type'
+            ]
+        )
+        config = Configuration(host=HOST, signing_info=signing_cfg)
+        # Set the OAuth2 acces_token to None. Here we are interested in testing
+        # the HTTP signature scheme.
+        config.access_token = None
+
+        api_client = petstore_api.ApiClient(config)
+        pet_api = petstore_api.PetApi(api_client)
+
+        mock_pool = MockPoolManager(self)
+        api_client.rest_client.pool_manager = mock_pool
+
+        mock_pool.expect_request('POST', 'http://localhost/v2/pet',
+                                 body=json.dumps(api_client.sanitize_for_serialization(self.pet)),
+                                 headers={'Content-Type': 'application/json',
+                                          'Authorization': 'Bearer ',
+                                          'User-Agent': 'OpenAPI-Generator/1.0.0/python'},
+                                 preload_content=True, timeout=TimeoutWithEqual(total=5))
+        mock_pool.expect_request('POST', 'http://localhost/v2/pet',
+                                 body=json.dumps(api_client.sanitize_for_serialization(self.pet)),
+                                 headers={'Content-Type': 'application/json',
+                                          'Authorization': 'Bearer ',
+                                          'User-Agent': 'OpenAPI-Generator/1.0.0/python'},
+                                 preload_content=True, timeout=TimeoutWithEqual(connect=1, read=2))
+
+        pet_api.add_pet(self.pet, _request_timeout=5)
```

Comment:
How does this check results from the mock _pool?
Can we verify that the signed data is eqal to a specific value?

Reply:
This was outdated. I initially copy-pasted the code from another file, but I wasn't ready to actually test. See my other comment, now I have several unit tests. I will probably add more tomorrow.

Reply:
I have added a bunch of unit tests.